### PR TITLE
Add map markers

### PR DESCRIPTION
**Pull request by caseycesari**
## Overview

Adds markers for subregions for the active region. See commit messages for details.

Connects #13 

## Demo

tnc1

## Testing Instructions

- Select a subregion. Verify that markers are added only for that subregion.
- Click a marker. Verify a popup is opened, and that the marker (real a copy of the marker) is placed above the popup window and acts as a popup tip.
- Click the "Open map" button in the popup and ensure that a new tab is opened to the selected region.
- Unactivate the region. Verify that the markers are removed.

### CODE REVIEW

File: src/js/views.js [hunk 1]
```diff
@@ -52,6 +71,51 @@ window.MapView = Backbone.View.extend({
...
+        var subregions = null;
+
+        if (this.model.get('selectedRegion')) {
+            subregions = this.model.get('selectedRegion').get('subregions');
+
+            if (subregions && subregions.length > 0) {
+                _.forEach(subregions, function(region) {
+                    if (region.mapCenter) {
+                        this.markers.addLayer(this.createMarker(region));
+                    }
```
Comment (kellyi) [inline src/js/views.js:90]: Wonder if it's worth adding an `else` here to `console.warn("Region's mapCenter is not configured properly.")`.

File: src/js/views.js [hunk 2]
```diff
@@ -52,6 +71,51 @@ window.MapView = Backbone.View.extend({
...
+            this.map.removeLayer(this.markers);
+            this.markers.clearLayers();
+        }
+    },
+
+    createMarker: function(region) {
+        var offset = [0, 25]; // Marker is used as popup tip
+        var popupContents = new MarkerPopupView({ model: region });
+        var marker = L.marker(region.mapCenter, { icon: this.icon })
+            .bindPopup(popupContents.render().el, { offset: offset });
```
Comment (kellyi) [inline src/js/views.js:105]: Do we need to manually destroy the `MarkerPopupView` objects when the markers are removed or will they be garbage collected when the markers are destroyed?

**Comment by kellyi** [comment]
These look great!

screen shot 2017-11-07 at 9 57 01 am

**Comment by kellyi** [comment]
This is mostly working well! In Safari and Chrome the "Open Map" links work correctly: they open each specific map site in a new tab.

Having some trouble getting the "Open Map" links to work in FF and IE11 though: clicking the "Open Map" button doesn't work as a link even though I can see the button press happening.

**Comment by caseycesari** [comment]
Thanks for the review. I'll take a look.

File: src/js/views.js (hunk 2 above)
Comment (caseycesari) [reply src/js/views.js:105]: Good question. I removed them manually to be safe in 6a6d4d4.

File: src/js/views.js (hunk 1 above)
Comment (caseycesari) [reply src/js/views.js:90]: I don't think it's worth it at the moment. I'd say we give them a chance to figure out the configuration, and if there are any issues, go from there.

**Comment by caseycesari** [comment]
Ready for another look.

**Review by kellyi — APPROVED** [review]
+1. Tested this out in IE11, FF, Chrome, and Safari and it works well in each!

**Comment by caseycesari** [comment]
Thanks for the review.

**Pull request merged**